lib/ecto/query/planner.ex

```diff
@@ -403,6 +404,13 @@ defmodule Ecto.Query.Planner do
     fields = subquery_source_fields(source)
     {keep_source_or_struct(source), subquery_fields(fields, ix)}
   end
+  defp subquery_select({:map, _, [{:&, _, [ix]}]}, take, query) do
+    fields = map_fields!(:select, query, ix)
+    take = Select.add_take(take, ix, {:map, fields})
+    {source, _} = source_take!(:select, query, take, ix, ix, %{})
+    fields = subquery_source_fields(source)
+    {keep_source_or_struct(source), subquery_fields(fields, ix)}
+  end
```

Comment:
I think for subqueries it doesn't matter. This can be treated exactly the same as `{:&, _, [ix]}` because we never return a struct or a map. Can you try something like:

```
  defp subquery_select({:map, _, [{:&, _, [_]} = entry]}, take, query) do
    subquery_select(entry, take, query)
  end
```

?

Reply:
I remember why I did this now. If you try this integration test:

```elixir
@tag :map_1
  test "map/1" do
    p = TestRepo.insert!(%Post{})
    c = TestRepo.insert!(%Comment{post_id: p.id})
    q = from p in Post, select: map(p)
    IO.inspect TestRepo.all(subquery(q))
  end
```

It will return make a struct because the `take` is not tagging the fields as `:map`.